File: java-wrapper/src/main/java/com/codedisaster/steamworks/SteamMatchmakingPingResponse.java
package com.codedisaster.steamworks;

public abstract class SteamMatchmakingPingResponse extends SteamInterface {

	protected SteamMatchmakingPingResponse() {
		super(~0L);
		callback = createProxy(this);
	}

	public abstract void serverResponded(SteamMatchmakingGameServerItem server);

	public abstract void serverFailedToRespond();

	// @off

	/*JNI
		#include "SteamMatchmakingPingResponse.h"
	*/

	private static native long createProxy(SteamMatchmakingPingResponse javaCallback); /*
		return (intp) new SteamMatchmakingPingResponse(env, javaCallback);
	*/

}

File: java-wrapper/src/main/native/SteamMatchmakingPingResponse.cpp
#include "SteamMatchmakingPingResponse.h"
#include "SteamMatchmakingGameServerItem.h"

SteamMatchmakingPingResponse::SteamMatchmakingPingResponse(JNIEnv* env, jobject callback)
	: SteamCallbackAdapter(env, callback) {

}

SteamMatchmakingPingResponse::~SteamMatchmakingPingResponse() {

}

void SteamMatchmakingPingResponse::ServerResponded(gameserveritem_t &server) {
	invokeCallback({
		jobject item = createGameServerItem(env, server);
		callVoidMethod(env, "serverResponded",
					   "(Lcom/codedisaster/steamworks/SteamMatchmakingGameServerItem;)V", item);
	});
}

void SteamMatchmakingPingResponse::ServerFailedToRespond() {
	invokeCallback({
		callVoidMethod(env, "serverFailedToRespond", "()V");
	});
}
